Fix style affectation on new line

simbas opened the pull request:
related to issue 15211: http://support.web-education.net/issues/15211
related to issue 19610: http://support.web-education.net/issues/19610

Proposed solution:
When I press enter in a styled node, the editor use cloning methods instead of straight forward affectation to style attribute.
When I apply new style properties at the end of a list, the editor is looking for the previous list item and move the selection to the end of that item.
When I press delete at the beginning of a line, the editor delete the line or the previous line based on the content of the lines.

simbas committed:
fix(editor): related to #15211 fix style affectation on new line

- fix the way style is affected to the newly created node when pressing
enter in a styled node
- adds Firefox to the targeted browsers of unit testing (the issue was
firefox only)

simbas committed:
fix(editor): fix the siblings tree position after pressing enter

simbas committed:
fix(editor): issue 19610 move range when at the end of a list

- move the range to the latest text node of the previous list item when
    the range select a text node at the end of the list

rdjedjig commented on src/ts/editor/onPressEnter.ts:
```diff
@@ -8,6 +8,23 @@ export function isElementNodeWithName(node: Node, nodeName: string): boolean {
     return isElementNode(node) && node.nodeName === nodeName;
 }
 
+export function toKebabCase(s: string): string {
```
Je connaissais pas ce nom !
L'implémentation n'est pas complète. (cf https://lodash.com/docs/4.17.10#kebabCase)
C'est pas très grave

rdjedjig replied:
y'a pas un classe utils pour ranger cette fonction ?

simbas replied:
J'ai ajouté un petit commentaire pour souligner que l'implémentation est partielle.

rdjedjig commented on src/ts/editor/onPressEnter.ts:
```diff
@@ -8,6 +8,23 @@ export function isElementNodeWithName(node: Node, nodeName: string): boolean {
+    return s.replace(/([a-z0-9])([A-Z])/g, '$1-$2').toLowerCase();
+}
+
+function copyStyle(dest: HTMLElement, src: HTMLElement) {
```
Dans quel cas propertyPriority est différent ou égal ? l'élément dest est toujours nouveau ?

simbas replied:
J'ai mergé #11 sur cette branche, cette fonction n'existe plus.

simbas committed:
docs(editor): adds a warning about toKebabCase implementation

- https://github.com/entcore/infra-front/pull/10#discussion_r227422478
- https://github.com/entcore/infra-front/pull/10#discussion_r227427185

simbas committed:
fix(editor): fix pressing enter at the start of a line

simbas committed:
fix(editor): delete zws char at the beginning of the line when press delete

simbas committed:
fix(editor): remove editor content when pressing enter while selecting the editor

simbas committed:
fix(editor): fix editor when pressing enter in a text which contains a <br>

simbas committed:
fix(editor): issue 19610, fix new line when in styled list

simbas committed:
test(editor): fix test compatibility with IE11

- also fix a little typo in onPressDelete tests

simbas committed:
fix(editor): fix editor when pressing enter at the beginning of editor content

simbas committed:
fix(editor): fix deletion while selecting

simbas committed:
fix(editor): fix previous line deletion when pressing delete at the beginning of a line
